src/runtime-rs/crates/hypervisor/src/device/vsock.rs

```diff
@@ -15,3 +20,55 @@ pub struct VsockConfig {
     /// unix domain socket path
     pub uds_path: String,
 }
+
+#[derive(Debug)]
+pub struct VsockConfig {
+    /// Unique identifier of the device
+    pub id: String,
+
+    /// A 32-bit Context Identifier (CID) used to identify the guest.
+    pub guest_cid: u32,
+
+    pub vhost_fd: File,
+}
+
+const VHOST_VIRTIO: u8 = 0xAF;
+nix::ioctl_write_ptr!(vhost_vsock_set_guest_cid, VHOST_VIRTIO, 0x60, u64);
+
+const CID_RETRY_TIMES: u32 = 50;
```

Comment:
Nit: I wonder if "count" would be clearer than "times":

```suggestion
const CID_RETRY_COUNT: u32 = 50;
```

Reply:
fixed